add QEMU Guest Agent

hgy59 opened the pull request:
## Description

- initial version of qemu-ga, builds for arch-x64-7.1 only

<!--Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.-->

Fixes #6019

## Checklist

- [x] Build rule `all-supported` completed successfully (only x64 on DSM 7 supported)
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

- [x] New Package

hgy59 committed:
add QEMU Guest Agent (#6019)

- initial version, builds for arch-x64-7.1 only

hgy59 commented:
@sgelineau17 can you please try, whether this works with proxmox?

r0bb10 commented:
> > did you follow some guide or are you able to write how you did? i was looking how to compile it for 7.1 and kvmx64.
> 
> @r0bb10 you find a package in the build of #6152

posting here instead of the other issue to keep it clean.

installed and tested on a virtualdsm instance, it does not want to run because of 

```
root@test:/volume1/@appstore/qemu-ga/bin# qemu-ga
1718832139.611087: critical: error opening channel '/dev/virtio-ports/org.qemu.guest_agent.0': No such file or directory
1718832139.611123: critical: failed to create guest agent channel
1718832139.611129: critical: failed to initialize guest agent channel

hgy59 commented:
@tofuHero did you try to add the missing folder `/var/packages/qemu/var/run` ? 

after creating the run folder, I get a different error (not running under qemu/proxmox):
```
$ sudo qemu-ga
1720716108.104297: critical: error opening channel '/dev/virtio-ports/org.qemu.guest_agent.0': No such file or directory
1720716108.104594: critical: failed to create guest agent channel
1720716108.104675: critical: failed to initialize guest agent channel
```

r0bb10 commented:
> @tofuHero did you try to add the missing folder `/var/packages/qemu/var/run` ?
> 
> after creating the run folder, I get a different error (not running under qemu/proxmox):
> 
> ```
> $ sudo qemu-ga
> 1720716108.104297: critical: error opening channel '/dev/virtio-ports/org.qemu.guest_agent.0': No such file or directory
> 1720716108.104594: critical: failed to create guest agent channel
> 1720716108.104675: critical: failed to initialize guest agent channel
> ```

i've done that and also got this error.

hgy59 committed:
add var/run folder

hgy59 commented:
@r0bb10 I did some analysis on the qemu-ga package of https://spk7.imnks.com/

1. qemu-ga runs as a service
2. the service must run as root, so you have to patch the `privilege` file after installation and each update (to replace package by root) and then restart the package
3. the package installs the var/run folder
4. the package includes two kernel modules for console (hvc_console.ko and virtio_console.ko)
5. before service start:
   - the package loads the console modules on demand with insmod
   - the package creates the virtio-ports device on demand (and binds /dev/vport1p1 to /dev/virtio-ports/org.qemu.guest_agent.0)
   - the package creates the folder /var/local/run on demand

We could clone the functions of the implementation by imnks.com/jim.plus, but I doubt that it will be more stable.

hgy59 committed:
run qemu-ga as service

- must run as root, so manual action is required for installation an upgrade
- ensure /dev/virtio-ports/org.qemu.guest_agent.0 exists before service starts

hgy59 commented:
@tofuHero, @r0bb10 I did some progress.
On my virtual DSM qemu-ga is running now as service (with manual patch of privilege file and manual service start).

I guess when not running virtual dsm, the kernel module `virtio_console.ko` is not running and we must include it in the package.

Can you please check with:
`lsmod | grep -i virtio_console`

on my virtual dsm it shows:
```
# lsmod | grep -i virtio_console
virtio_console         19522  1
virtio_ring             5789  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
virtio                  5698  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
```

r0bb10 commented:
> @tofuHero, @r0bb10 I did some progress. On my virtual DSM qemu-ga is running now as service (with manual patch of privilege file and manual service start).
> 
> I guess when not running virtual dsm, the kernel module `virtio_console.ko` is not running and we must include it in the package.
> 
> Can you please check with: `lsmod | grep -i virtio_console`
> 
> on my virtual dsm it shows:
> 
> ```
> # lsmod | grep -i virtio_console
> virtio_console         19522  1
> virtio_ring             5789  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
> virtio                  5698  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
> ```

mm installed the latest version Build #6811, done the run-as root, package starts and no error seem obvious

```
root@vdsm:/volume1/@appstore/qemu-ga/bin# /volume1/@appstore/qemu-ga/bin/qemu-ga -v
1720894497.519524: debug: Guest agent version 9.0.1 started
1720894497.519566: debug: disabling command: guest-get-devices
1720894497.519635: debug: received EOF
1720894497.620480: debug: received EOF
1720894497.721515: debug: received EOF
1720894497.822490: debug: received EOF
1720894497.923571: debug: received EOF
1720894498.024563: debug: received EOF
1720894498.125515: debug: received EOF
1720894498.226596: debug: received EOF
1720894498.327556: debug: received EOF
1720894498.428602: debug: received EOF
1720894498.529475: debug: received EOF
1720894498.630571: debug: received EOF
1720894498.731513: debug: received EOF
1720894498.832537: debug: received EOF
1720894498.933481: debug: received EOF
1720894499.034596: debug: received EOF
```

and

```
root@vdsm:/volume1/@appstore/qemu-ga/bin# lsmod | grep -i virtio_console
virtio_console         19522  1
virtio_ring             5789  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
virtio                  5698  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
```

tbc0309 commented:
#6019 add more

xutongtongxue commented:
Hi @hgy59  I'm a xpenology user and also waiting for this package from SynoCommunity, I can see there is pull request from you but it has not been merged yet. may I know when the package will be available in the repository? Thanks for you effort for this package.

sgelineau17 commented:
> Hi @hgy59 I'm a xpenology user and also waiting for this package from SynoCommunity, I can see there is pull request from you but it has not been merged yet. may I know when the package will be available in the repository? Thanks for you effort for this package.

I'm in same the case, I waiting this package from SynoCommunity too.

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_qemu

hgy59 committed:
update wizard files

r0bb10 commented:
with the latest push still shows same error:

```
root@virtualdsm:/volume1/@appstore/qemu-ga/bin# ./qemu-ga -v
1733853669.949037: debug: Guest agent version 9.0.1 started
1733853669.949079: debug: disabling command: guest-get-devices
1733853669.949158: debug: received EOF
1733853670.049638: debug: received EOF
1733853670.150609: debug: received EOF
1733853670.251617: debug: received EOF
1733853670.352593: debug: received EOF
1733853670.453567: debug: received EOF
1733853670.554584: debug: received EOF
1733853670.655641: debug: received EOF
1733853670.756608: debug: received EOF
1733853670.857595: debug: received EOF
1733853670.958583: debug: received EOF
1733853671.059501: debug: received EOF
1733853671.160601: debug: received EOF
1733853671.261656: debug: received EOF
1733853671.362617: debug: received EOF
1733853671.463610: debug: received EOF
1733853671.564622: debug: received EOF
1733853671.665661: debug: received EOF
1733853671.766663: debug: received EOF
root@virtualdsm:/volume1/@appstore/qemu-ga/bin# lsmod | grep -i virtio_console
virtio_console         19522  1
virtio_ring             5789  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi
virtio                  5698  6 virtio_blk,virtio_net,virtio_pci,virtio_rng,virtio_console,virtio_scsi

hgy59 committed:
qemu update from v9.0.1 to v9.1.2

xutongtongxue commented:
@hgy59  Hi, I just downloaded the latest x64-7.1 artifact from the Actions and tried to install to my synology VM, but the installation failed. My DSM version is 7.2.1-69057, I want to know if this package supports DSM 7.2.1-69057, so I can help test this package. Thanks.

r0bb10 commented:
@hgy59 still no go with 9.1.2

```
root@virtualdsm:/volume1/@appstore/qemu-ga/bin# ./qemu-ga -v
1733905920.599879: debug: Guest agent version 9.1.2 started
1733905920.599994: debug: received EOF
1733905920.700570: debug: received EOF
1733905920.801571: debug: received EOF
1733905920.902575: debug: received EOF
1733905921.003670: debug: received EOF
1733905921.104664: debug: received EOF
1733905921.205585: debug: received EOF
1733905921.306532: debug: received EOF
1733905921.407592: debug: received EOF
1733905921.508484: debug: received EOF
1733905921.609617: debug: received EOF
1733905921.710613: debug: received EOF
1733905921.811618: debug: received EOF
1733905921.912557: debug: received EOF
1733905922.013548: debug: received EOF

sgelineau17 commented:
> > > Hi @hgy59 I'm a xpenology user and also waiting for this package from SynoCommunity, I can see there is pull request from you but it has not been merged yet. may I know when the package will be available in the repository? Thanks for you effort for this package.
> > 
> > 
> > @xutongtongxue, @sgelineau17 can one confirm that this package works as expected?
> 
> for me still not working.

Hello,
I don't know how to install this package and my DSM are in the last release (DSM 7.2.2-72806 Update 2) and my Proxmox too

mreid-tt commented:
@r0bb10, I'm not familiar with this but would the scenario posted [here](https://forum.proxmox.com/threads/alpine-linux-kvm-wont-shutdown-with-qemu-guest-agent.49866/#post-232549) be relevant to the issue you are experiencing?

r0bb10 commented:
> > Hello,
> > I don't know how to install this package and my DSM are in the last release (DSM 7.2.2-72806 Update 2) and my Proxmox too.
> > Thanks
> 
> 1. download the artifacts file for x64-7.1 from github build action (current is here https://github.com/SynoCommunity/spksrc/actions/runs/12262090338/artifacts/2301545743) (don't worry DSM 7.1 package is for DSM 7.2 too)
> 2. extract the spk file from the zip archive
> 3. manually install the spk in DSM package center
> 4. you have to ssh into DSM and patch the privileges file as documented in the install wizard page
> 5. I guess you have to enable serial driver/console/port (?) as reported here [Qemu Guest Agent for Synology VM on Proxmox #6019 (comment)](https://github.com/SynoCommunity/spksrc/issues/6019#issuecomment-2250022879)

still no go adding serial0 (i already had serial2 enabled, adding 0 suggested in the link did do nothing :( )
dunno if it's related to run on virtualdsm but it's wierd, it should work.

any check i can do to help debugging?

xutongtongxue commented:
> > > Hello,
> > > I don't know how to install this package and my DSM are in the last release (DSM 7.2.2-72806 Update 2) and my Proxmox too.
> > > Thanks
> > 
> > 
> > 
> > 1. download the artifacts file for x64-7.1 from github build action (current is here https://github.com/SynoCommunity/spksrc/actions/runs/12262090338/artifacts/2301545743) (don't worry DSM 7.1 package is for DSM 7.2 too)
> > 2. extract the spk file from the zip archive
> > 3. manually install the spk in DSM package center
> > 4. you have to ssh into DSM and patch the privileges file as documented in the install wizard page
> > 5. I guess you have to enable serial driver/console/port (?) as reported here [Qemu Guest Agent for Synology VM on Proxmox #6019 (comment)](https://github.com/SynoCommunity/spksrc/issues/6019#issuecomment-2250022879)
> 
> still no go adding serial0 (i already had serial2 enabled, adding 0 suggested in the link did do nothing :( ) dunno if it's related to run on virtualdsm but it's wierd, it should work.
> 
> any check i can do to help debugging?

The latest package in the skp7 repository requires the serial console to be removed from PVE, have you tried removing the serial console to see if the package works? Thanks.

xutongtongxue commented:
@hgy59 @r0bb10  I tried installing the package(change the privilege and start a service), the PVE now can get the IP from the qemu-ga and the shutdown command from PVE also works fine. Thanks.

![image](https://github.com/user-attachments/assets/c998b699-7176-4763-8a25-58f23b9d4b11)
![image](https://github.com/user-attachments/assets/7ef23a84-5ca3-48e3-a1ed-73f0a9cdbbda)

When try to click shutdown from PVE web console, the result of 'tail -f /volume1/@appdata/qemu-ga/qemu-ga.log' are as follows. 

![image](https://github.com/user-attachments/assets/cba58552-5152-429a-84c8-8eba9979353c)

Below is my env.
```
# PVE 8.2.9
# Synology VM DSM 7.2.1-69057
# Serial0 added into Synology VM
```

r0bb10 commented:
@xutongtongxue we are using different enviroments, you are on xpenology i'm on virtualdsm.

@hgy59 i see on the latest screenshots posted that qemu-ga shows the docker container stopping via serial0, i can see this happen on serial2 on my side on virtualdsm, is it possible to set qemu-ga to listen to serial2?

xutongtongxue commented:
@sgelineau17 @tofuHero  please try this, I have installed the this package and it works fine for me. Below is my env. Thanks.

```
# PVE version: 8.2.9
# Synology loader: RR V22.24.1.1
# Synology version: DSM 7.2.1-69057
```

> > Hello,
> > I don't know how to install this package and my DSM are in the last release (DSM 7.2.2-72806 Update 2) and my Proxmox too.
> > Thanks
> 
> 1. download the artifacts file for x64-7.1 from github build action (current is here https://github.com/SynoCommunity/spksrc/actions/runs/12262090338/artifacts/2301545743) (don't worry DSM 7.1 package is for DSM 7.2 too)
> 2. extract the spk file from the zip archive
> 3. manually install the spk in DSM package center
> 4. you have to ssh into DSM and patch the privileges file as documented in the install wizard page
> 5. I guess you have to enable serial driver/console/port (?) as reported here [Qemu Guest Agent for Synology VM on Proxmox #6019 (comment)](https://github.com/SynoCommunity/spksrc/issues/6019#issuecomment-2250022879)

xutongtongxue commented:
> @xutongtongxue we are using different enviroments, you are on xpenology i'm on virtualdsm.
> 
> @hgy59 i see on the latest screenshots posted that qemu-ga shows the docker container stopping via serial0, i can see this happen on serial2 on my side on virtualdsm, is it possible to set qemu-ga to listen to serial2?

@r0bb10  Sorry I didn't know your issue or the backaground of issue.  I just check the flows again and still did't get your issue. Can you explain your requirements and problems again. Perhaps, I can help you.

denmalley commented:
Just wanted to pop in to comment that I was able to get this to work with a newly installed xpenology system running dsm 7.2.2 under proxmox.  Had already installed the one showing in the repository, but followed the steps posted by @hgy59 above.  It overwrote/replaced the one I had already installed and all is working well!

grimandreas commented:
@r0bb10, @hgy59,
By default, Virtual DSM has it's own serial port for getting info about serial, license, etc on /dev/vport1p1.
So, qemu guest agent on Virtual DSM creates second vport - /dev/vport2p1 and after package installation it's necessary to edit file **/usr/syno/etc/packages/qemu-ga** and set virtio serial port for qemu-ga manually:
`SERVICE_COMMAND="${DAEMON} --daemonize --logfile ${LOG_FILE} --pidfile ${PID_FILE} -p /dev/vport2p1"`
For me, after than change, qemu-ga works like a charm on VirtualDSM.

r0bb10 commented:
> @r0bb10, @hgy59, By default, Virtual DSM has it's own serial port for getting info about serial, license, etc on /dev/vport1p1. So, qemu guest agent on Virtual DSM creates second vport - /dev/vport2p1 and after package installation it's necessary to edit file **/usr/syno/etc/packages/qemu-ga** and set virtio serial port for qemu-ga manually: `SERVICE_COMMAND="${DAEMON} --daemonize --logfile ${LOG_FILE} --pidfile ${PID_FILE} -p /dev/vport2p1"` After that change, qemu-ga works like a charm on Virtual DSM.

yeah this is what i [thought ](https://github.com/SynoCommunity/spksrc/pull/6152#issuecomment-2538009970) but i did not thinker with the start script to set a different listening port, i set it up now like you suggested and it finally works.

i had to edit line 26 in `/var/packages/qemu-ga/scripts/service-setup` , restarting the package now proxmox is happy.

![image](https://github.com/user-attachments/assets/d7452220-62b5-4dba-83a8-c29532c1826c)

not that it really matters since it already worked with acpi commands but it's nice to have the guest agent fully work.

hgy59 commented:
@grimandreas, @r0bb10 shall we introduce a wizard variable to define the virtual port (default `/dev/vport1p1`) at installation time? If we have such a variable, the setting will not be lost at package updates.

Or shall we try to autodetect the port at service start?

r0bb10 commented:
> @grimandreas, @r0bb10 shall we introduce a wizard variable to define the virtual port (default `/dev/vport1p1`) at installation time? If we have such a variable, the setting will not be lost at package updates.
> 
> Or shall we try to autodetect the port at service start?

the problem is only present on a virtualdsm instance on proxmox, which is probably a very niece scenario since not many use virtualdsm on non-synology hardware.. and the vm has to be setup correctly, including the snippet companion.

but i agree that could be very handy to have an advanced option in the wizard to redirect to port2 (with a big warning asking if the package is being installed on a virtualdsm and correctly configured).. i guess autodetect would not work in a scenario where the vm is virtualdsm but no serial2 has been setup.

grimandreas commented:
@r0bb10, 
> i guess autodetect would not work in a scenario where the vm is virtualdsm but no serial2 has been setup.

In vanilla Virtual DSM running on DSM hardware /dev/vport1p1 also exists. As I told previously, it's used for retriving from host information about license, s/n, cpu information, etc. In real DSM hardware first serial port exists too. 

@hgy59, 
> shall we introduce a wizard variable to define the virtual port (default `/dev/vport1p1`) at installation time? If we have such a variable, the setting will not be lost at package updates.
> 
> Or shall we try to autodetect the port at service start?

I don't know if we could really autodetect the port at service start. Wizard variable could be optimal in this case (with default port value `/dev/vport1p1`).

sgelineau17 commented:
Hello, 
How to install it now ?  The artecfact is not available today.
Thanks

hgy59 commented:
> How to install it now ? The artecfact is not available today.

Is is still here (available for 90 days, use the latest workflow)
https://github.com/SynoCommunity/spksrc/actions/runs/12262090338?pr=6152#artifacts

hgy59 committed:
qemu-ga update and enhance installation wizard

- update qemu-ga to v9.2.0
- add option to configure alternate serial port
- remove created virtio port at uninstall

hgy59 commented:
@r0bb10 qemu-ga is now updated to v9.2.0 and has an option in the installation wizard to define an alternate serial port.

r0bb10 commented:
> @r0bb10 qemu-ga is now updated to v9.2.0 and has an option in the installation wizard to define an alternate serial port.

![image](https://github.com/user-attachments/assets/dd03fb10-9bcb-41f5-a55e-9bba8c426c47)

confirm that wizard works, great!

grimandreas commented:
@hgy59 , thanks for that changes! I can also confirm that new build works perfectly. I set custom port in wizard, so `/var/packages/qemu-ga/var/custom_port.txt` now contains:
![image](https://github.com/user-attachments/assets/4cf94963-2bb0-4159-a713-050602bec755)
All qemu-ga functions seem to work as well.

nightmare28 commented:
I have still issue to start qemu-ga, event do the sudo sed....

d4rkw1n9 commented:
> > How to install it now ? The artecfact is not available today.
> 
> It is still here (available for 90 days, use the latest workflow) https://github.com/SynoCommunity/spksrc/actions/runs/12262090338?pr=6152#artifacts

Probably just expired :-/ Any recent workflow on that or where could the file be potentially downloaded please? :) Thanks!

hgy59 commented:
> Probably just expired :-/ Any recent workflow on that or where could the file be potentially downloaded please? :) Thanks!

Since this PR has the label "status/published" you can find the package on https://synocommunity.com/package/qemu-ga or in DSM Package Center when you added the SynoCommunity Repository.

d4rkw1n9 commented:
Okay, thanks a lot!

Mar 31, 2025 10:38:16 AM hgy59 ***@***.***>:

> *DuckDuckGo* removed one tracker. _More[https://duckduckgo.com/-1n2qN1s_5OZkvJNbR4Sl6-GcKT8hg65A4UsDKPcwdVP0CbPOUxlrW5ZFY2IRFm2gsEukwALE4N-tGWTQIQhyWW7vz27pYa6N7gboWjK7z7f1si-qtsZzL8CtaHzdnNRIQsGAEgTcnbsbJsK64y3MawC2kW385gk8ETt3DzMjEDMieSrTf0yjO8sKepD-6Ke3FqOM-kXlu9MTnuTYfvrSoVONF6QmBMYANTVCT-4j6fsEp5_ZBbWPgt4Qtg9MZMrAL9TWDM7F0tJ0rjeaFQJ23MWz8xPMnPgk4YO1cLmNVBlVifiScUjFj-Mk8PGKXcamdv5zxw8cmdILJeSaq32r1W-ZZNA1w2qXe-mcu93I9MPXbAlTQv6rxAuuC2cjPLFdan55eNsiJaL3Ch-WoqxkEYDi0_zD5jHeux4gpZrtFVltfx3t4DisvZ0oTUw9ewRJRpV7KjEoavd00Pv1dxZc9H2v0xYEmA7LTvDp7-eqafmly1f5wZPMT4W7MZKr4WsxRE16NgY]_
> Report Spam[https://duckduckgo.com/-1n2qN1s_5OZkvJNbR4Sl6-GcKT8hg65A4UsDKPcwdVP0CbPOUxlrW5ZFY2IRFm2gsEukwALE4N-tGWTQIQhyWW7vz27pYa6N7gboWjK7z7f1si-qtsZzL8CtaHzdnNRIQsGAEgTcnbsbJsK64y3MawC2kW385gk8ETt3DzMjEDMieSrTf0yjO8sKepD-6Ke3FqOM-kXlu9MTnuTYfvrSoVONF6QmBMYANTVCT-4j6fsEp5_ZBbWPgt4Qtg9MZMrAL9TWDM7F0tJ0rjeaFQJ23MWz8xPMnPgk4YO1cLmNVBlVifiScUjFj-Mk8PGKXcamdv5zxw8cmdILJeSaq32r1W-ZZNA1w2qXe-mcu93I9MPXbAlTQv6rxAuuC2cjPLFdan55eNsiJaL3Ch-WoqxkEYDi0_zD5jHeux4gpZrtFVltfx3t4DisvZ0oTUw9ewRJRpV7KjEoavd00Pv1dxZc9H2v0xYEmA7LTvDp7-eqafmly1f5wZPMT4W7MZKr4WsxRE16NgY]
> 
> 
> 
>      [Image][hgy59][https://avatars.githubusercontent.com/u/12009964?s=20&v=4]*hgy59* left a comment (SynoCommunity/spksrc#6152)[https://github.com/SynoCommunity/spksrc/pull/6152#issuecomment-2765515480]
>     
> 
> Probably just expired :-/ Any recent workflow on that or where could the file be potentially downloaded please? :) Thanks!
> 
> Since this PR has the label "status/published" you can find the package on https://synocommunity.com/package/qemu-ga or in DSM Package Center when you added the SynoCommunity Repository.
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/SynoCommunity/spksrc/pull/6152#issuecomment-2765515480], or unsubscribe[https://github.com/notifications/unsubscribe-auth/BFKW5A4SU2HBI4CN2HYUVUD2XD5HPAVCNFSM6AAAAABJSVRDNSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRVGUYTKNBYGA].
> You are receiving this because you commented.
>

ilium007 commented:
```
xxx@xpenology01:/volume1/@appdata/qemu-ga$ cat qemu-ga.log
Sun May 11 22:26:48 AEST 2025
Begin service_prestart
End service_prestart
Starting qemu-ga command /volume1/@appstore/qemu-ga/bin/qemu-ga --daemonize --logfile /volume1/@appdata/qemu-ga/qemu-ga.log --pidfile /volume1/@appdata/qemu-ga/qemu-ga.pid
1746966408.935384: critical: error opening channel '/dev/virtio-ports/org.qemu.guest_agent.0': Device or resource busy
1746966408.935402: critical: failed to create guest agent channel
1746966408.935407: critical: failed to initialize guest agent channel
```

grimandreas commented:
@ilium007, just do `ls -lah /dev | grep vport*` inside your VM and see the output. If I'm right you will see two serial ports:
![image](https://github.com/user-attachments/assets/6876a2db-11c1-4ac0-833e-c22dec60eaf8)
If so, you will need to set custom serial port in `/var/packages/qemu-ga/var/custom_port.txt` and then restart service.

ilium007 commented:
No, already checked that. Only one virtual serial device:

```
xxxx@xpenology01:~$ ll /dev/vport*
crw------- 1 root root 248, 1 May 11 23:43 /dev/vport0p1
```

ilium007 commented:
So.... this is the reason. qemu-ga is already running. I have installed DSM via Arc Loader - they must have qemu-ga deployed somehow.

```
xxxx@xpenology01:~$ ps aux | grep qemu
root     14453  0.0  0.0  86616  2524 ?        Ssl  10:35   0:00 /usr/vmtools/bin/qemu-ga -m virtio-serial -p /dev/virtio-ports/org.qemu.guest_agent.0 -t /var/run/ -f /var/run/vmtools.pid
```
